spec/services/hearings_creator_spec.rb

```diff
@@ -250,56 +250,51 @@
-
-        expect(Sqs::PublishHearing).to receive(:call).once.with(hash_including(shared_time: "2018-10-25 11:30:00",
-                                                                               jurisdiction_type: "MAGISTRATES",
-                                                                               case_urn: "12345",
-                                                                               defendant: defendant_one,
-                                                                               court_centre_id: "dd22b110-7fbc-3036-a076-e4bb40d0a519"))
-        expect(Sqs::PublishHearing).to receive(:call).once.with(hash_including(shared_time: "2018-10-25 11:30:00",
-                                                                               jurisdiction_type: "MAGISTRATES",
-                                                                               case_urn: "12345",
-                                                                               defendant: defendant_two,
-                                                                               court_centre_id: "dd22b110-7fbc-3036-a076-e4bb40d0a519"))
+        LaaReference.create!(defendant_id: "dd22b110-7fbc-3036-a076-e4bb40d0a666", linked: true, maat_reference: "123", user_name: "Bob")
+        LaaReference.create!(defendant_id: "ad22b110-7fbc-3036-a076-e4bb40d0a667", linked: true, maat_reference: "456", user_name: "Steve")
+
+        expect(Sqs::PublishHearing).to receive(:call).twice.with(hash_including(shared_time: "2018-10-25 11:30:00",
+                                                                                jurisdiction_type: "MAGISTRATES",
+                                                                                case_urn: "12345",
+                                                                                defendant: master_defendant,
+                                                                                court_centre_id: "dd22b110-7fbc-3036-a076-e4bb40d0a519"))
         create_hearings
       end
     end
 
-    context "with two defendants one with linked LAA reference, one without" do
+    context "with two defendants, one with linked LAA reference, one without" do
       let(:prosecution_case_array) { nil }
       let(:defendant_case_array) { [defendant_case_one, defendant_case_two] }
 
-      it "calls the Sqs::PublishHearing service twice" do
-        allow(LaaReference).to receive(:find_by).with(defendant_id: "dd22b110-7fbc-3036-a076-e4bb40d0a666", linked: true).and_return(defendant_one)
-        allow(LaaReference).to receive(:find_by).with(defendant_id: "ad22b110-7fbc-3036-a076-e4bb40d0a667", linked: true).and_return(nil)
+      it "calls the Sqs::PublishHearing service once" do
+        LaaReference.create!(defendant_id: "dd22b110-7fbc-3036-a076-e4bb40d0a666", linked: true, maat_reference: "123", user_name: "Bob")
```

Comment:
So this creates a real LaaReference and puts it into the db. Is there any way to set up the test without doing this? Like stubbing `LaaReference.find_by` and setting it to return the boolean we want for the test. I don't like the calling of the real thing in the tests when it's then depending on a second class working properly

Reply:
Sorry, what don't you like about it?